megablocks/layers/glu_test.py

```diff
@@ -0,0 +1,74 @@
+import unittest
+from functools import partial
+
+from absl.testing import parameterized
+from megablocks.layers.arguments import Arguments
+from megablocks.layers import dmoe
+from megablocks.layers import testing
+import torch
+
+
+def test_modules(
+        hidden_size,
+        ffn_hidden_size,
+        moe_num_experts=1,
+        moe_capacity_factor=1,
+        moe_top_k=1,
+        use_grouped_gemm=False):
+    init_method = partial(torch.nn.init.normal_, mean=0.0, std=0.1)
+    args = Arguments(
+        hidden_size=hidden_size,
+        ffn_hidden_size=ffn_hidden_size,
+        moe_num_experts=moe_num_experts,
+        moe_capacity_factor=moe_capacity_factor,
+        moe_top_k=moe_top_k,
+        init_method=init_method,
+        memory_optimized_mlp=False,
+        mlp_type='glu',
+        use_grouped_gemm=use_grouped_gemm,
+        fp16=False,
+        bf16=True)
+
+    glu = testing.GLU(args)
+    dmoe_glu = dmoe.dMoE(args)
```

Comment:
I think you can actually simplify this and just test against the MLP classes? I think it is ok to assume those don't interact with the dMoE outer logic to simplify the unit tests. We're only running the version where there is only one expert anyways, I think.

Reply:
I gave this a go, but wasn't totally sure how to best instantiate the `topo` topology matrix used in `SpaseMLP.forward` in a minimal way outside of using `dMoe`. I was able to get the grouped mlp working. Is there a one to two-liner that could set this up easily? (otherwise I can make a helper `_topology` function in the test to set that matrix up). 
...
This is the snippet of code from the test, basically I'm unclear on how to set up the topology matrix, in the sparseGLU case:

``` x = torch.randn(sl, bs, hs).to(torch.bfloat16).cuda()

        _, glu, dmoe_glu = test_modules(
            hidden_size=hs,
            ffn_hidden_size=hs * 2,
            grouped_mlp=grouped_mlp)

        expected_out = glu(x)

        tokens_per_expert = torch.tensor([x.shape[2]]).cuda()
        if grouped_mlp:
            out = dmoe_glu(x.view(-1, hs), tokens_per_expert)
        else:
            topology = < how to best do this>
            out = dmoe_glu(x.view(-1, hs), topology)

        out = out.view(sl, bs, hs)

Reply:
Ah ya, that's a pain. What you can do is actually just create a random sparse matrix using STK, I think. Like you did for the `mul` op in STK.